doc/source/whatsnew/v1.2.0.rst

```diff
@@ -289,6 +291,7 @@ Deprecations
 - Deprecated :meth:`Index.is_all_dates` (:issue:`27744`)
 - Deprecated automatic alignment on comparison operations between :class:`DataFrame` and :class:`Series`, do ``frame, ser = frame.align(ser, axis=1, copy=False)`` before e.g. ``frame == ser`` (:issue:`28759`)
 - :meth:`Rolling.count` with ``min_periods=None`` will default to the size of the window in a future version (:issue:`31302`)
+- Using "outer" ufuncs on DataFrames to return 4d ndarray is now deprecated. Convert to an ndarray first (:issue:`23743`)
```

Comment:
where does this happen? a test for this?

Reply:
```python
In [3]: df = pd.DataFrame({"A": [1, 2]})

In [4]: np.add.outer(df, df)
Out[4]:
array([[[[2],
         [3]]],


       [[[3],
         [4]]]])
```

This matches a deprecation / removal we did for Series. Tested at https://github.com/pandas-dev/pandas/pull/36955/files#diff-f47ac75f060a28193618c0eb4ff98a4eR108